src_base/xeve_param_parse.c

```diff
@@ -0,0 +1,309 @@
+/* Copyright (c) 2022, Samsung Electronics Co., Ltd.
+   All Rights Reserved. */
+/*
+   Redistribution and use in source and binary forms, with or without
+   modification, are permitted provided that the following conditions are met:
+
+   - Redistributions of source code must retain the above copyright notice,
+   this list of conditions and the following disclaimer.
+
+   - Redistributions in binary form must reproduce the above copyright notice,
+   this list of conditions and the following disclaimer in the documentation
+   and/or other materials provided with the distribution.
+
+   - Neither the name of the copyright owner, nor the names of its contributors
+   may be used to endorse or promote products derived from this software
+   without specific prior written permission.
+
+   THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+   AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+   IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+   ARE DISCLAIMED.IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+   LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+   CONSEQUENTIAL DAMAGES(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+   SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+   INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+   CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+   ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+   POSSIBILITY OF SUCH DAMAGE.
+*/
+
+#include "xeve_param_parse.h"
+
+#include "xeve.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stddef.h>
+
+#define PARAMS_END_KEY                      (0)
+#define PARAM_STR_MAX_LEN                   (256)
+
+#define OFFSET(x) offsetof(XEVE_PARAM, x)
+
+#define SET_XEVE_PARAM_METADATA( param, data_type) \
+    { .name=#param, .type=data_type, .offset=OFFSET(param) }
+
+/* Type of data stored by a given param from XEVE_PARAM stuct */
+typedef enum DataType {
```

Comment:
Can you replace these variable names to ALL CAPS - underscore format?

Reply:
of course, I can